### DISCUSSION TITLE
Support Adaptive Design

### DISCUSSION THREAD

**Discussion by MohammadPCh** [vercel/next.js#24540] [Ideas]
### Feature Request
Hi :) I would like to use Adaptive design instead of responsive design in NextJS.
When i audit my NextJS website with lighthouse (for mobile devices) it said you have **unused** CSS and JavaScript.
and also said your JavaScript main thread is large and recommend to decrease it.
So I think if I can serve different components for mobile and desktop i can handle it.

**Comment by MohammadPCh** [comment]
I need to serve different components for same URL.

**Comment by cryptiklemur** [comment]
Would love something like this, where the Data Fetching patterns would each have access to a parameter like "isMobile" and "isDesktop".
